### PR TITLE
remove py2

### PR DESCRIPTION
**Pull request by jacobbogdanov**
TODO: 
- [ ] still testing

### CODE REVIEW

File: platforms/BUILD.bazel [hunk 1]
```diff
@@ -14,12 +14,11 @@ package(default_visibility = ["//visibility:public"])
 # Create a config setting for each 2-tuple of (python version, system platform).
```
Comment (ian-h-chamberlain) [inline platforms/BUILD.bazel:14]: nit: stale comment

File: python/pytest/rules.bzl [hunk 2]
```diff
@@ -41,54 +41,41 @@ def pytest_test(
...
-        )
+    pytest_deps = [
+        "@pip3//pytest",
+        "@rules_128tech//rules_128tech/pytest_plugins:pytest_bazel_sharder",
+    ]
+
+    pytest_deps.extend([
+        "@pip3//pytest_timeout",
+        "@pip3//pdbpp",
+    ])
```
Comment (ian-h-chamberlain) [inline python/pytest/rules.bzl:54]: nit: just put these in the first `pytest_deps` definition

File: thirdparty/pip/README.md [hunk 3]
```diff
@@ -134,7 +134,7 @@ $(bazel info output_base)/external/pip<major>/
 
 Where:
 
-`<major>` is the major Python version for which the problem is occurring (indicated by `@pip2` or `@pip3`)
+`<major>` is the major Python version for which the problem is occurring (indicated by `@pip3` or `@pip3`)
```
Comment (ian-h-chamberlain) [inline thirdparty/pip/README.md:137]: Probably would just remove or rewrite this sentence (and other references to `<major>` or similar)

**Review by ian-h-chamberlain — CHANGES_REQUESTED** [review]
🙌

Some minor documentation / find+replace updates needed (requesting changes for those) but mostly LGTM!

File: thirdparty/pip/README.md (hunk 3 above)
Comment (jacobbogdanov) [reply thirdparty/pip/README.md:137]: yeah I noticed all the copy-paste errors in this file while upgrading our internal deps... we have this same README in it's entirety there too!

**Review by ian-h-chamberlain — APPROVED** [review]

**Pull request merged**